### PR TITLE
[Meson] Initial commit of meson build

### PR DESCRIPTION
**Pull request by ohsewon**
Initial commit of meson. This commit's goal is to make single binary.

Signed-off-by: sewon.oh <sewon.oh@samsung.com>

---
# [Template] PR Description

In general, github system automatically copies your commit message for your convenience.
Please remove unused part of the template after writing your own PR description with this template.
```bash
$ git commit -s filename1 filename2 ... [enter]

Summarize changes in around 50 characters or less

More detailed explanatory text, if necessary. Wrap it to about 72
characters or so. In some contexts, the first line is treated as the
subject of the commit and the rest of the text as the body. The
blank line separating the summary from the body is critical;
various tools like `log`, `shortlog` and `rebase` can get confused 
if you run the two together.

Further paragraphs come after blank lines.

**Changes proposed in this PR:**
- Bullet points are okay, too
- Typically a hyphen or asterisk is used for the bullet, preceded
  by a single space, with blank lines in between, but conventions vary here.

Resolves: #123
See also: #456, #789

**Self evaluation:**
1. Build test: [ ]Passed [ ]Failed [*]Skipped
2. Run test: [ ]Passed [ ]Failed [* ]Skipped

**How to evaluate:**
1. Describe how to evaluate in order to be reproduced by reviewer(s).

Add signed-off message automatically by running **$git commit -s ...** command.

$ git push origin <your_branch_name>
```

### CODE REVIEW

File: meson.build [hunk 1]
```diff
@@ -0,0 +1,55 @@
+project('nnstreamer', 'c', 'cpp',
```
Comment (leemgs) [inline meson.build]: Could you append how to use Meson in ./Documentation/ folder?

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    version: '0.0.3',
+    license: ['LGPL'],
+    meson_version: '>=0.29.0',
```
Comment (leemgs) [inline meson.build]: As well all know, Meson runs based on the Ninja package. Note that we have to append "BuildRequires: ninja" additionally for the Meson build in ./packaging/ (for Tizen) and ./debian/ (for Ubuntu).

Comment (leemgs) [inline meson.build]: > '>=0.29.0',

Currently, Most of the gstreamer packages and gstreamer-based tools require >=0.47. 
As we all know, NNstreamer is based on Gstreamer. Please check which number do we have to prepare the minimum version of Meson to build NNStreamer in advance.

File: meson.build [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    gstreamer_audio
+]
+
+conf = configuration_data()
+conf.set('VERSION', meson.project_version())
+# configure_file(output: 'config.h', configuration: conf)
+
+inc = include_directories('./gst/nnstreamer')
+
+projects = [
```
Comment (leemgs) [inline meson.build]: The project list looks good to me at least. :smile:

File: meson.build (hunk 1 above)
Comment (ohsewon) [reply meson.build]: I will update next commit.

File: meson.build (hunk 2 above)
Comment (ohsewon) [reply meson.build]: It is right. change version to >= 0.47.0

File: gst/tensor_converter/meson.build [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+]
+
+tensor_converter = library('tensor_converter',
+    tensor_converter_sources,
+    dependencies: [
+        dependencies
+    ],
+    include_directories: inc,
+    link_with: tensor_common
+)
```
Comment (leemgs) [inline gst/tensor_converter/meson.build]: Need to fix a newline issue.

**Comment by ohsewon** [comment]
Build with meson and gst-launch test.

screenshot from 2018-11-16 17-12-09

**Pull request merged**